pkg/plugin/plugin.go

```diff
@@ -50,15 +51,44 @@ func (r *RpcPlugin) SetWeight(rollout *v1alpha1.Rollout, desiredWeight int32, ad
 			ErrorString: err.Error(),
 		}
 	}
+
 	if gatewayAPIConfig.HTTPRoute != "" {
-		return r.setHTTPRouteWeight(rollout, desiredWeight, additionalDestinations, &gatewayAPIConfig)
+		gatewayAPIConfig.HTTPRoutes = append(gatewayAPIConfig.HTTPRoutes, gatewayAPIConfig.HTTPRoute)
 	}
+
 	if gatewayAPIConfig.TCPRoute != "" {
-		return r.setTCPRouteWeight(rollout, desiredWeight, additionalDestinations, &gatewayAPIConfig)
+		gatewayAPIConfig.TCPRoutes = append(gatewayAPIConfig.TCPRoutes, gatewayAPIConfig.TCPRoute)
 	}
-	return pluginTypes.RpcError{
-		ErrorString: GatewayAPIManifestError,
+
+	if len(gatewayAPIConfig.HTTPRoutes) == 0 && len(gatewayAPIConfig.TCPRoutes) == 0 {
+		return pluginTypes.RpcError{
+			ErrorString: GatewayAPIManifestError,
+		}
 	}
+
+	for _, name := range gatewayAPIConfig.HTTPRoutes {
+		err := r.setHTTPRouteWeight(rollout, desiredWeight, additionalDestinations, &GatewayAPITrafficRouting{
+			Namespace: gatewayAPIConfig.Namespace,
+			HTTPRoute: name,
+		})
+		if err.HasError() {
+			return pluginTypes.RpcError{
+				ErrorString: fmt.Sprintf("set weight for HTTPRoute %q: %s", name, err),
+			}
+		}
```

Comment:
https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-gatewayapi/pull/34#discussion_r1480056376
To be sure that it is seen, Github decided it is oudated